### PR TITLE
Create envproxy-only chart

### PR DESCRIPTION
**Pull request by deansheather**
# !!! DO NOT MERGE THIS PR !!!

This PR exists only for comments/review and diff. It should never be merged as it's basically a fork.

This is what customers will use (via coder-cli) in 1.15 to deploy a non-local alpha workspace provider to another K8s cluster. This will be tidied and replaced by a neat subchart when workspace providers hits beta.

### Changes

- Removes cemanager, dashboard, timescale
- Removes unnecessary templates for RBAC stuff
- Removes unnecessary values
- Removes cemanager/dashboard from ingress rules
- Removes `CODER_SERVER_USER` and `CODER_SERVER_KEY` env vars from envproxy
- Important envproxy vars:
    - `ENVPROXY_ACCESS_URL`:  URL to report to cemanager, taken from `envproxy.accessURL` or derived from `ingress`
    - `SSH_ENABLED`: `true` or `false` to report to cemanager
    - `CEMANAGER_ACCESS_URL`: the full URL used to connect to cemanager, minus `/api`
    - `CEMANAGER_TOKEN`: the envproxy's token
    - `NAMESPACE_WHITELIST`: the namespace whitelist, comma separated (to report to cemanager)
    - `DEVURL_DOMAIN`: the dev URL domain (also reported to cemanager)

Devurls TLS can be sniffed from the envproxy access URL (assume TLS is enabled if envproxy has TLS).

### CODE REVIEW

File: templates/_common.tpl [hunk 1]
```diff
@@ -99,11 +56,19 @@ tolerations:
...
+    {{- if .Values.ingress.tls.enable -}}
+    https://
+    {{- else -}}
+    http://
+    {{- end -}}
+    {{- .Values.ingress.host }}/ingress
+{{- else }}
+{{- fail "envproxy.access.URL or ingress.host must be set" }}
+{{- end }}
+{{- end }}
```
Comment (f0ssel) [inline templates/_common.tpl:74]: Does this implicitly not support raw IPs?

Comment (f0ssel) [reply templates/_common.tpl:74]: If so, I think that's OK for alpha? But we will need to in the future, may have to poll kube for it on the ingress from envproxy itself or something on start.

File: templates/envproxy.yaml [hunk 2]
```diff
@@ -128,11 +119,94 @@ spec:
...
+{{ $namespaces := append .Values.namespaceWhitelist .Release.Namespace }}
+{{- range $namespaces }}
+---
+# The envproxy role lists the permissions the envproxy requires.
+# It conditionally sets the pod security policy based on the
+# helm value.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: envproxy
```
Comment (f0ssel) [inline templates/envproxy.yaml:140]: shouldn't need this anymore

File: values.yaml [hunk 3]
```diff
@@ -68,92 +65,16 @@ devurls:
...
-      cpu: "250m"
-      memory: "512Mi"
-
 # envproxy contains configuration for the service handling long-lived
 # connections to environments such as IDE or shell sessions.
 envproxy:
+  # envproxy.accessURL -- The URL reported to cemanager. Must be accessible by
+  # cemanager and all users who can use this workspace provider. Derived from
+  # ingress.host if not set. This should be a full URL, complete with protocol
+  # and trailing "/ingress" (no trailing slash).
```
Comment (f0ssel) [inline values.yaml:76]: what does `/ingress` mean here?

File: values.yaml [hunk 4]
```diff
@@ -182,55 +103,14 @@ envproxy:
...
-  # e.g.
-  # tolerations:
-  # - key: "my key"
-  #   operator: "Exists"
-  #   effect: "NoExecute"
-  #   tolerationSeconds: 6000
-  tolerations: []
+# cemanager -- Details for connecting to the cemanager instance.
+# NOTE: the envproxy must be accessible by envproxy.accessURL from the
+# cemanager.
```
Comment (f0ssel) [inline values.yaml:108]: should this note be on the envproxy accessURL instead? And here we say "cemanager must be accessible"

File: templates/_common.tpl (hunk 1 above)
Comment (deansheather) [reply templates/_common.tpl:74]: Yeah to support the raw IP of the load balancer you'd have to do it like that. Alternatively, the user could deploy with a fake envproxy access URL, wait for load balancer IP, and redeploy with real IP.

File: values.yaml (hunk 3 above)
Comment (deansheather) [reply values.yaml:76]: oops I meant `/proxy`

**Comment by deansheather** [comment]
Closing in favor of #20

**Pull request closed without merging**